torch_geometric/explain/algorithm/subgraphX_explainer.py

```diff
@@ -0,0 +1,646 @@
+                # here we check the resulting sub-graphs and
+                # only keep the largest one
+                subgraph_coalition = [
+                    node for node in all_nodes if node != each_node
+                ]
+
+                subgraphs = [
+                    self.graph.subgraph(c) for c in nx.connected_components(
+                        self.graph.subgraph(subgraph_coalition))
+                ]
+
+                if self.new_node_idx:
+                    for sub in subgraphs:
+                        if self.new_node_idx in list(sub.nodes()):
+                            main_sub = sub
+                else:
+                    main_sub = subgraphs[0]
+
+                    for sub in subgraphs:
+                        if sub.number_of_nodes() > main_sub.number_of_nodes():
+                            main_sub = sub
+
+                new_graph_coalition = sorted(list(main_sub.nodes()))
+
+                # check the state map and merge the same sub-graph
+                find_same = False
+                for old_graph_node in self.state_map.values():
+                    if Counter(old_graph_node.coalition) == Counter(
+                            new_graph_coalition):
+                        new_node = old_graph_node
+                        find_same = True
+
+                if not find_same:
+                    new_node = self.MCTSNodeClass(new_graph_coalition)
+                    self.state_map[str(new_graph_coalition)] = new_node
+
+                find_same_child = False
+                for cur_child in tree_node.children:
+                    if Counter(cur_child.coalition) == Counter(
+                            new_graph_coalition):
+                        find_same_child = True
+
+                if not find_same_child:
+                    tree_node.children.append(new_node)
+
+            scores = compute_scores(self.score_func, tree_node.children)
+            for child, score in zip(tree_node.children, scores):
+                child.P = score
+
+        sum_count = sum([c.N for c in tree_node.children])
+        selected_node = max(tree_node.children,
+                            key=lambda x: x.Q() + x.U(sum_count))
+        v = self.mcts_rollout(selected_node)
+        selected_node.W += v
+        selected_node.N += 1
+        return v
+
+    def mcts(self, verbose=True):
+        if verbose:
+            print(f"The nodes in graph is {self.graph.number_of_nodes()}")
+        for rollout_idx in range(self.n_rollout):
+            self.mcts_rollout(self.root)
+            if verbose:
+                print(f"At the {rollout_idx} rollout, {len(self.state_map)}"
+                      "states that have been explored.")
+
+        explanations = [node for _, node in self.state_map.items()]
+        explanations = sorted(explanations, key=lambda x: x.P, reverse=True)
+        return explanations
+
+
+class SubgraphXExplainer(ExplainerAlgorithm):
```

Comment:
Should everything above this be moved into a separate file? Personally I'm unsure.